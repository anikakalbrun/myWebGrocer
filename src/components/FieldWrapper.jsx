import React from 'react';
import {string, bool, node} from 'prop-types';

FieldWrapper.propTypes = {
    label: string,
    touched: bool,
    error: string,
    children: node
};

FieldWrapper.defaultProps = {};

export default function FieldWrapper({children, label, touched, error}) {
    return (
        <div className="form-group">
            <label className="col-xs-12">{label}:</label>
            <div className="col-xs-12">
                {children}
                {touched && error ?
                    <div>
                        <div className='has-error'/>
                        <small className="error">
                            <strong>{label}:</strong> {error}</small>
                    </div>
                    :
                    <div className='no-error'/>
                }
            </div>
        </div>
    );
}