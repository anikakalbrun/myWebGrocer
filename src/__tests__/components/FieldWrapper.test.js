import React from 'react';
import FieldWrapper from '../../components/FieldWrapper';

let wrapper;

describe('FieldWrapper', () => {

    it('renders correctly when no error is present', () => {
        const props = {
            children: <input name='firstName'/>,
            label: "First Name",
            touched: false,
            error: undefined
        };
        wrapper = renderer.create(<FieldWrapper {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('does not render error when field was not touched', () => {
        const props = {
            children: <input name='firstName'/>,
            label: "First Name",
            touched: false,
            error: 'required'
        };
        wrapper = renderer.create(<FieldWrapper {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders error message when error is present and field was touched', () => {
        const props = {
            children: <input name='firstName'/>,
            label: "First Name",
            touched: true,
            error: 'required'
        };
        wrapper = renderer.create(<FieldWrapper {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
