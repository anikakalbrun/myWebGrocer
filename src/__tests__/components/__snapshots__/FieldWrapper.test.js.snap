// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWrapper does not render error when field was not touched 1`] = `
<div
  className="form-group"
>
  <label
    className="col-xs-12"
  >
    First Name
    :
  </label>
  <div
    className="col-xs-12"
  >
    <input
      name="firstName"
    />
    <div
      className="no-error"
    />
  </div>
</div>
`;

exports[`FieldWrapper renders correctly when no error is present 1`] = `
<div
  className="form-group"
>
  <label
    className="col-xs-12"
  >
    First Name
    :
  </label>
  <div
    className="col-xs-12"
  >
    <input
      name="firstName"
    />
    <div
      className="no-error"
    />
  </div>
</div>
`;

exports[`FieldWrapper renders error message when error is present and field was touched 1`] = `
<div
  className="form-group"
>
  <label
    className="col-xs-12"
  >
    First Name
    :
  </label>
  <div
    className="col-xs-12"
  >
    <input
      name="firstName"
    />
    <div>
      <div
        className="has-error"
      />
      <small
        className="error"
      >
        <strong>
          First Name
          :
        </strong>
         
        required
      </small>
    </div>
  </div>
</div>
`;
